packages/core/admin/server/bootstrap.js

```diff
@@ -52,6 +53,22 @@ const syncAuthSettings = async () => {
   await adminStore.set({ key: 'auth', value: newAuthSettings });
 };
 
+const syncAPITokensPermissions = async () => {
+  const validPermissions = strapi.contentAPI.permissions.providers.action.keys();
+  const permissionsInDB = await pipeAsync(
+    strapi.query('admin::api-token-permission').findMany,
+    map('action')
+  )();
+
+  const unknownPermissions = uniq(difference(permissionsInDB, validPermissions));
+
+  if (unknownPermissions.length > 0) {
+    await strapi
+      .query('admin::api-token-permission')
+      .deleteMany({ where: { action: { $in: unknownPermissions } } });
```

Comment:
Same bulk update problem as elsewhere. This doesn't clean up the _links table.

Reply:
It does on my side, though :thinking: 
Did you test it on your side?

Reply:
You're right. That's very odd since it worked differently for me elsewhere, but good to hear. I need to look deeper into the bulk functions and find out exactly what each one does and doesn't do regarding relations.